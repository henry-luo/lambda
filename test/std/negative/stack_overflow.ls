// Test: Stack Overflow (Should Crash)
// Category: negative
// Type: negative
// Expected: error

// This test should cause a stack overflow
let recursive_bomb = fn(n) {
    recursive_bomb(n + 1)
}

recursive_bomb(0)
