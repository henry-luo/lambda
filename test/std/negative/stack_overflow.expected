09:15:53 [NOTICE] Start transpiling test/std/negative/stack_overflow.ls...
[script:list]
  [content:list[2]]
    no declare
    item:
    [let stam:null]
      declare:
      [assign expr:recursive_bomb:function]
        [function expr:function]
          params:
          [param: n:any]
          [primary expr:any,const:0]
            [call expr:any]
              [primary expr:any,const:0]
                [ident:recursive_bomb:any,const:0]
              args:
              [binary expr +.3:any]
                [primary expr:any,const:0]
                  [ident:n:any,const:0]
                [primary expr:int,const:1]
                  [const: int]
    item:
    [primary expr:any,const:0]
      [call expr:any]
        [primary expr:function,const:0]
          [ident:recursive_bomb:function,const:0]
        args:
        [primary expr:int,const:1]
          [const: int]
09:15:53 [NOTICE] compiling C code in 'test/std/negative/stack_overflow.ls' to MIR
C2MIR init end           -- 68 usec
test/std/negative/stack_overflow.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2438, parse tokens -- 1464
  C2MIR preprocessor end    -- 495 usec
  C2MIR parser end          -- 618 usec
test/std/negative/stack_overflow.ls:452:42: warning -- using pointer without cast for integer type parameter
  C2MIR context checker end -- 704 usec
  C2MIR generator end       -- 749 usec
C2MIR compiler end                -- 782 usec
09:15:53 [NOTICE] Generating native code...
##### Script 'test/std/negative/stack_overflow.ls' executed: #####
null
